File: app/src/main/java/com/example/notes/NoteDao.java
package com.example.notes;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Single;

@Dao
interface NoteDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Completable insertNote(Note note);

    @Query("select * from notes_table ORDER BY id DESC")
    Single<List<Note>> getNotes();

    @Query("delete from notes_table where id = :id")
    Completable deleteNote(int id);
}

File: app/src/main/java/com/example/notes/NoteDao.kt
package com.example.notes

import android.widget.EditText

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

import io.reactivex.Completable
import io.reactivex.Single

@Dao
internal interface NoteDao {

    @get:Query("select * from notes_table ORDER BY id DESC")
    val notes: Single<List<Note>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertNote(note: Note): Completable

    @Query("delete from notes_table where id = :id")
    fun deleteNote(id: Int): Completable
}
